client/state/posts/selectors.js

```diff
@@ -362,6 +359,61 @@ export const isEditedPostDirty = createSelector(
 );
 
 /**
+ * Returns true if the post status is publish, private, or future
+ * and the date is in the past
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Number}  postId Post ID
+ * @return {Boolean}        Whether post is published
+ */
+export const isPostPublished = createSelector(
+	( state, siteId, postId ) => {
+		const post = getSitePost( state, siteId, postId );
+
+		if ( ! post ) {
+			return null;
+		}
+
+		return includes( [ 'publish', 'private' ], post.status ) ||
+			( post.status === 'future' && moment( post.date ).isBefore( moment() ) );
+	},
+	( state ) => state.posts.items
+);
+
+/**
+ * Returns the slug, or suggested_slug, for the edited post
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @param  {Number} postId Post ID
+ * @return {String}             Slug value
+ */
+export function getEditedPostSlug( state, siteId, postId ) {
+	const post = getSitePost( state, siteId, postId );
```

Comment:
Rather than generating all these values up-front, since we're sometimes returning earlier than they're used, we could reorganize the function to be a bit more efficient:

```js
export function getEditedPostSlug( state, siteId, postId ) {
	// if local edits exists, return them regardless of post status
	const postEdits = getPostEdits( state, siteId, postId );
	if ( has( postEdits, 'slug' ) ) {
		return postEdits.slug;
	}

	const post = getSitePost( state, siteId, postId );
	const postSlug = get( post, 'slug' );

	// when post is published, return the slug
	if ( isPostPublished( state, siteId, postId ) ) {
		return postSlug;
	}

	// only return suggested_slug if slug has not been edited
	const suggestedSlug = get( post, [ 'other_URLs', 'suggested_slug' ] );
	if ( suggestedSlug && ! postSlug ) {
		return suggestedSlug;
	}

	return postSlug;
}
```